openedx/core/djangoapps/api_admin/tests/test_models.py

```diff
@@ -69,3 +73,41 @@ def test_unicode(self):
             unicode(ApiAccessConfig(enabled=False)),
             u'ApiAccessConfig [enabled=False]'
         )
+
+
+class ApiAccessRequestSignalTests(TestCase):
+    def setUp(self):
+        super(ApiAccessRequestSignalTests, self).setUp()
+        self.user = UserFactory()
+        self.api_access_request = ApiAccessRequest(user=self.user, base_url="http://example.com")
+
+    def test_save_signal_success_email(self):
+        send_new_pending_email_function = 'openedx.core.djangoapps.api_admin.models._send_new_pending_email'
+        send_decision_email_function = 'openedx.core.djangoapps.api_admin.models._send_decision_email'
+
+        # Verify that
+        with mock.patch(send_new_pending_email_function) as mock_new_email:
+            self.api_access_request.save()
+        mock_new_email.assert_called_once_with(self.api_access_request)
+
+        self.api_access_request.status = ApiAccessRequest.APPROVED
+
+        with mock.patch(send_decision_email_function) as mock_decision_email:
+            self.api_access_request.save()
```

Comment:
You could do `self.api_access_request.approve()` and delete line 93 -- I'd rather keep the status-changing operations behind the model's API when we can.